gcloud-java-core/src/main/java/com/google/gcloud/Service.java

```diff
@@ -16,6 +16,6 @@
 
 package com.google.gcloud;
 
-public interface Service<O extends ServiceOptions> {
-  O options();
+public interface Service<OptionsT extends ServiceOptions<?, OptionsT>> {
```

Comment:
Is it crazy to throw aside the "made-at-Google" thing and evaluate what things would look like if we had started outside of Google? I'm guessing several projects still use Google-style, I'm just curious how many...

(To put this into perspective, Google-style for Python is not PEP8, but we follow PEP8 for gcloud-python...)